tests/unit/withOnyxTest.js

```diff
@@ -101,6 +101,34 @@ describe('withOnyx', () => {
             });
     });
 
+    it('should replace arrays inside objects with withOnyx subscribing to individual key if mergeCollection is used', () => {
+        const collectionItemID = 1;
+        const TestComponentWithOnyx = withOnyx({
+            text: {
+                key: `${ONYX_KEYS.COLLECTION.TEST_KEY}${collectionItemID}`,
+            },
+        })(ViewWithCollections);
+        const onRender = jest.fn();
+        render(<TestComponentWithOnyx onRender={onRender} />);
+        return waitForPromisesToResolve()
+            .then(() => {
+                Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {
+                    test_1: {list: [1, 2]},
+                });
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                Onyx.mergeCollection(ONYX_KEYS.COLLECTION.TEST_KEY, {
+                    test_1: {list: [7]},
+                });
+                return waitForPromisesToResolve();
+            })
+            .then(() => {
+                expect(onRender.mock.calls.length).toBe(3);
```

Comment:
OH! I also had doubts about this. Not sure why it is 3 times, I was kind of expecting only 2. Here's what `onRender.mock.calls` shows:
```
      [
        {
          onRender: [Function: mockConstructor] {
            _isMockFunction: true,
          },
          collections: {},
          testObject: { isDefaultProp: true }
        },
        {
          onRender: [Function: mockConstructor] {
          },
          text: { list: [ 1, 2 ] },
          collections: {},
          testObject: { isDefaultProp: true }
        },
        {
          onRender: [Function: mockConstructor] {
            _isMockFunction: true,
          },
          text: { list: [7] },
          collections: {},
          testObject: { isDefaultProp: true }
        }
      ]
```
Not sure what the first call is, I assume it's expected?

Reply:
Ah the first subscription always will trigger a render since it's like:

- withOnyx waits for data (there is none for this key)
- withOnyx renders child (1)
- data changes once and component renders again (2)
- data changes once more and component renders again (3)

Reply:
If this test works (looks like it) then I am 👍 on not updating the `setState` call in `keysChanged()`

Reply:
Ah, got it, thanks for the explanation!